fuel/datasets/hdf5.py

```diff
@@ -113,25 +115,105 @@ class H5PYDataset(Dataset):
     """
     ref_counts = defaultdict(int)
 
-    def __init__(self, path, which_set=None, subset=None, load_in_memory=False,
+    def __init__(self, path, which_set, subset=None, load_in_memory=False,
                  flatten=None, driver=None, **kwargs):
         self.path = path
+        self.driver = driver
+        if which_set not in self.available_splits:
+            raise ValueError(
+                "'{}' split is not provided by this ".format(which_set) +
+                "dataset. Available splits are " +
+                "{}.".format(self.available_splits))
         self.which_set = which_set
-        self.subset = subset if subset else slice(None, None, None)
+        subset = subset if subset else slice(None)
+        if subset.step not in (1, None):
+            raise ValueError("subset.step must be either 1 or None")
+        self.subsets = [subset for source in self.provides_sources]
         self.load_in_memory = load_in_memory
         self.flatten = [] if flatten is None else flatten
-        self.driver = driver
-
-        super(H5PYDataset, self).__init__(**kwargs)
-
         for source in self.flatten:
             if source not in self.provides_sources:
-                raise ValueError("trying to flatten source " +
-                                 "'{}' which is not part of ".format(source) +
-                                 "the provided sources")
+                raise ValueError(
+                    "trying to flatten source '{}' which is ".format(source) +
+                    "not provided by the '{}' split".format(self.which_set))
+
+        super(H5PYDataset, self).__init__(**kwargs)
 
         self.load()
 
+    @staticmethod
+    def create_split_array(split_dict):
+        """Create a valid array for the `split` attribute of the root node.
+
+        Parameters
+        ----------
+        split_dict : dict
+            Maps split names to dict. Those dict map source names to
```

Comment:
I don't mention it anywhere, but you're right, there is no such assumption.

There are some assumptions being made in the utility functions, but that's mostly a result of me not wanting to cover every imaginable case at the same time.

Would you like me to put that somewhere?